Task 2

supermaciz opened the pull request:
This PR implements Task 2, adding the ability for the assistant to fetch and display stock market data from the Alpha Vantage API using OpenAI's function calling capabilities.

supermaciz committed:
Add stock analysis assistant script and update dependencies

- Added `main.py` to create and use a stock analyzer assistant powered by OpenAI.
- Updated `requirements.txt` to include `python-dotenv`.
- Updated `.gitignore` to include `.idea/`.

supermaciz committed:
Update message content in thread creation for clarity

supermaciz committed:
Refactor thread creation and execution logic in `main.py` for better modularity

supermaciz committed:
Refactor `main.py` to improve function signatures and consolidate `dotenv` initialization

supermaciz committed:
Modularize stock analyzer implementation: split `main.py` into multiple modules with improved structure. Add Alpha Vantage integration.

supermaciz committed:
Task 2 ok

supermaciz committed:
Validate required parameters in `get_time_series` method of `alpha_vantage.py` and handle `interval` for intraday queries

nikalinaogorodova commented on stock_analyzer/__main__.py:
I'm not sure why `stock_analyzer` was made into a separate package (this is not necessary), and it's better to keep the logic in the `main.py` without packaging in this way

nikalinaogorodova commented on stock_analyzer/openai_ops.py:
```diff
@@ -0,0 +1,121 @@
+from .alpha_vantage import functions_list, name_to_function
+
+
+def create_client():
```
`dotenv.load_dotenv()` has disappeared from this task completely (but was present in the first task). Is this expected?

supermaciz committed:
Refactor: centralize API key management in config module

https://github.com/hyperskill-content/stock-analyzer/pull/4#discussion_r2211823819

supermaciz committed:
Fix: move pprint call under requires_action check to avoid unnecessary execution

Fix https://github.com/hyperskill-content/stock-analyzer/pull/5#discussion_r2213266077
